File: rb/features.rb
# VogelwarteSchweiz SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module VogelwarteSchweizFeatures
  def self.make_feature(name)
    case name
    when "base"
      VogelwarteSchweizBaseFeature.new
    when "ratelimit"
      VogelwarteSchweizRatelimitFeature.new
    when "retry"
      VogelwarteSchweizRetryFeature.new
    when "test"
      VogelwarteSchweizTestFeature.new
    when "timeout"
      VogelwarteSchweizTimeoutFeature.new
    else
      VogelwarteSchweizBaseFeature.new
    end
  end
end

File: py/features.py
# VogelwarteSchweiz SDK feature factory

from feature.base_feature import VogelwarteSchweizBaseFeature
from feature.test_feature import VogelwarteSchweizTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VogelwarteSchweizBaseFeature(),
        "test": lambda: VogelwarteSchweizTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
